app/Services/UserService.php

```diff
@@ -21,4 +21,19 @@ public function store(array $data): User
         // Create a new user using the provided data
         return User::create($data);
     }
+
+    public function update(User $user, array $data)
+    {
+        $user->fill($data);
+
+        if (isset($data['password'])) {
```

Comment:
you have already done : `$user->fill($data);` so the password is already set. Remove it